prov/util/src/util_ns.c

```diff
@@ -285,15 +285,16 @@ static void *util_ns_name_server_func(void *args)
 	}
 
 	for (p = res; p; p = p->ai_next) {
-		listenfd = ofi_socket(p->ai_family, p->ai_socktype, p->ai_protocol);
+		listenfd = ofi_socket(p->ai_family, p->ai_socktype,
+				      p->ai_protocol);
 		if (listenfd != INVALID_SOCKET) {
 			n = 1;
-			setsockopt(listenfd, SOL_SOCKET, SO_REUSEADDR,
-				   &n, sizeof(n));
-			if (!bind(listenfd, p->ai_addr, p->ai_addrlen))
-				break;
-			ofi_close_socket(listenfd);
-			listenfd = INVALID_SOCKET;
+			if (setsockopt(listenfd, SOL_SOCKET,
+				       SO_REUSEADDR, &n, sizeof(n)) ||
+			    (bind(listenfd, p->ai_addr, p->ai_addrlen))) {
+				ofi_close_socket(listenfd);
+				listenfd = INVALID_SOCKET;
+			}
 		}
```

Comment:
The change breaks the name server. The loop would continue even if bind succeed and in the end we got invalid listenfd unless the succeeded bind happens to the last one. 

Massive fabtests failures have been observed over psm and psm2 with this patch.

I suggest we revert this PR and take a differet approach for the Coverity issue. 

@shefty @gladkovdmitry17